File: frontend/src/api/candidateCenter.ts
import apiClient from './client'

export type CandidateCenterSort = 'updated_at_desc' | 'created_at_desc'

export interface CandidateCenterListQuery {
  assigned?: boolean
  status?: string
  pipeline_status?: string
  job_id?: string
  keyword?: string
  page?: number
  page_size?: number
  sort?: CandidateCenterSort
}

export interface CandidateCenterListItem {
  application_id: string
  candidate_id: string
  name: string
  phone?: string | null
  email?: string | null
  job_id: string
  job_name: string
  job_code: string
  job_version_id: string
  job_version_label?: string | null
  status: string
  pipeline_status: string
  round_id?: string | null
  round_name?: string | null
  sequence_no?: number | null
  round_status?: string | null
  schedule_status: string
  invitation_status: string
  transcript_status: string
  question_status: string
  analysis_status: string
  analysis_overall_score?: number | string | null
}

export interface CandidateCenterListResponse {
  items: CandidateCenterListItem[]
  total: number
  page: number
  page_size: number
}

export interface ScoreDimensionSummary {
  name: string
  weight: number
  score: number
}

export interface ScoreSummary {
  result_id: string
  version_label: string
  total_score: number
  calculated_total_score: number
  score_band: string
  recommendation: string
  summary: string
  information_insufficient: boolean
  is_stale: boolean
  is_current: boolean
  dimensions: ScoreDimensionSummary[]
}

export interface ResumeSummary {
  resume_id: string
  resume_version_id: string
  version_label: string
  kind: string
  status: string
  original_filename?: string | null
  confirmed_at?: string | null
}

export interface CandidateCenterRound {
  application_id: string
  round_id: string
  name: string
  sequence_no: number
  status: string
  schedule_status: string
  has_meeting_password: boolean
  invitation_status: string
  transcript_status: string
  question_status: string
  analysis_status: string
  analysis_overall_score?: number | string | null
}

export interface OtherApplicationSummary {
  application_id: string
  job_id: string
  job_name: string
  job_code: string
  status: string
  pipeline_status: string
  created_at: string
}

export interface CandidateCenterDetail {
  application_id: string
  candidate_id: string
  name: string
  phone?: string | null
  email?: string | null
  job_id: string
  job_name: string
  job_code: string
  job_version_id: string
  job_version_label?: string | null
  status: string
  pipeline_status: string
  close_action?: string | null
  interview_started: boolean
  resume_summary?: ResumeSummary | null
  score_summary?: ScoreSummary | null
  rounds: CandidateCenterRound[]
  other_applications: OtherApplicationSummary[]
}

const DEFAULT_LIST_QUERY = {
  assigned: true,
  page: 1,
  page_size: 20,
} as const

export async function listCandidateCenterApplications(
  params: CandidateCenterListQuery = {},
) {
  const { data } = await apiClient.get<CandidateCenterListResponse>(
    '/candidate-center/applications',
    {
      params: {
        ...DEFAULT_LIST_QUERY,
        ...params,
      },
    },
  )
  return data
}

export async function getCandidateCenterApplicationDetail(
  candidateId: string,
  applicationId: string,
) {
  const { data } = await apiClient.get<CandidateCenterDetail>(
    `/candidate-center/candidates/${candidateId}/applications/${applicationId}`,
  )
  return data
}

File: backend/app/schemas/candidate_center.py
from __future__ import annotations

from datetime import datetime
from decimal import Decimal
from typing import Literal
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field, field_validator

from app.models.candidate import APPLICATION_STATUSES
from app.models.resume import PIPELINE_STATUSES

_QUERY_CONFIG = ConfigDict(
    extra="forbid",
    str_strip_whitespace=True,
    hide_input_in_errors=True,
)


class CandidateCenterListQuery(BaseModel):
    model_config = _QUERY_CONFIG

    assigned: bool = True
    status: str | None = None
    pipeline_status: str | None = None
    job_id: UUID | None = None
    keyword: str | None = None
    page: int = Field(default=1, ge=1)
    page_size: int = Field(default=20, ge=1, le=100)
    sort: Literal["updated_at_desc", "created_at_desc"] = "updated_at_desc"

    @field_validator("status")
    @classmethod
    def _status_whitelist(cls, value: str | None) -> str | None:
        if value is None or value == "":
            return None
        if value not in APPLICATION_STATUSES:
            raise ValueError("invalid status")
        return value

    @field_validator("pipeline_status")
    @classmethod
    def _pipeline_whitelist(cls, value: str | None) -> str | None:
        if value is None or value == "":
            return None
        if value not in PIPELINE_STATUSES:
            raise ValueError("invalid pipeline_status")
        return value


class ScoreDimensionSummary(BaseModel):
    name: str
    weight: float
    score: float


class ScoreSummaryOut(BaseModel):
    result_id: UUID
    version_label: str
    total_score: float
    calculated_total_score: float
    score_band: str
    recommendation: str
    summary: str
    information_insufficient: bool
    is_stale: bool
    is_current: bool
    dimensions: list[ScoreDimensionSummary] = Field(default_factory=list)


class ResumeSummaryOut(BaseModel):
    resume_id: UUID
    resume_version_id: UUID
    version_label: str
    kind: str
    status: str
    original_filename: str | None = None
    confirmed_at: datetime | None = None


class CandidateCenterListItem(BaseModel):
    application_id: UUID
    candidate_id: UUID
    name: str
    phone: str | None = None
    email: str | None = None
    job_id: UUID
    job_name: str
    job_code: str
    job_version_id: UUID
    job_version_label: str | None = None
    status: str
    pipeline_status: str
    round_id: UUID | None = None
    round_name: str | None = None
    sequence_no: int | None = None
    round_status: str | None = None
    schedule_status: str
    invitation_status: str
    transcript_status: str
    question_status: str
    analysis_status: str
    analysis_overall_score: Decimal | None = None


class CandidateCenterListResponse(BaseModel):
    items: list[CandidateCenterListItem]
    total: int
    page: int
    page_size: int


class CandidateCenterRoundOut(BaseModel):
    application_id: UUID
    round_id: UUID
    name: str
    sequence_no: int
    status: str
    schedule_status: str
    has_meeting_password: bool = False
    invitation_status: str
    transcript_status: str
    question_status: str
    analysis_status: str
    analysis_overall_score: Decimal | None = None


class OtherApplicationSummary(BaseModel):
    application_id: UUID
    job_id: UUID
    job_name: str
    job_code: str
    status: str
    pipeline_status: str
    created_at: datetime


class CandidateCenterDetailOut(BaseModel):
    application_id: UUID
    candidate_id: UUID
    name: str
    phone: str | None = None
    email: str | None = None
    job_id: UUID
    job_name: str
    job_code: str
    job_version_id: UUID
    job_version_label: str | None = None
    status: str
    pipeline_status: str
    close_action: str | None = None
    interview_started: bool
    lock_version: int
    resume_summary: ResumeSummaryOut | None = None
    score_summary: ScoreSummaryOut | None = None
    rounds: list[CandidateCenterRoundOut] = Field(default_factory=list)
    other_applications: list[OtherApplicationSummary] = Field(default_factory=list)
